pkg/repo/badgerdb/repo.go

```diff
@@ -11,6 +11,7 @@ import (
 // Repo implements the repo interface.
 type Repo struct {
 	db *badger.DB
+	tx *badger.Txn
```

Comment:
I believe we can use a single field as the interface instead of a concrete type. Can you check what interfaces are defined in the `badger` and `sql`?

E.g., it does not matter whether we use `badger.Txn` or `badger.DB` in the `db` field.